ldt/src/ldt_based_pcs.rs

```diff
@@ -110,7 +111,10 @@ where
     type Error = L::Error;
 
     fn commit_batches(&self, polynomials: Vec<In>) -> (Self::Commitment, Self::ProverData) {
```

Comment:
The current signature does not specify any coset shift. We can then possibly maintain the `commit_shifted_batches` as is, and add a another method to deal with batch coset shifts. In this way, we avoid having to allocate a new `shifts` vector, to just repeat `1`s over the size of the `polynomials` `Vec`.

Reply:
Yes, the allocation is not ideal, but maintaining two functions is potentially problematic, no? any change in the commit function would have to be done in both functions. 

Other options are:
1. Accepting `IntoIterator<Item: Borrow<usize>>` instead of a slice, then we could use `iter::repeat`. 
2. Using `1` as a default value if the shift slice is shorter than the polynomial vector (this might lead to incorrect usage if someone doesn't align lengths correctly).  

Happy to implement any of these, and also separating completely to two functions.

Reply:
I prefer `1.`, another option is to create an enum to contain either a slice or a single value, and match against these two possibilities. In this way, it is possible to keep the current logic of `commit_batches` simpler

Reply:
I agree (1) seems reasonable, but IMO it should also be okay to just keep the allocation, most provers will only call this a few times and `polynomials.len()` should be small